app/lib/scorecard/backend.dart

```diff
@@ -25,6 +25,13 @@ final _logger = Logger('pub.scorecard.backend');
 
 final Duration _deleteThreshold = const Duration(days: 182);
 
+/// The maximum number of keys we'll try to lookup when we need to load the
+/// scorecard or the report information for multiple versions.
+///
+/// The Datastore limit is 1000, but that caused resource constraint issues
+/// https://github.com/dart-lang/pub-dev/issues/4040
+const _batchLookupMaxKeyCount = 400;
```

Comment:
I would suggest 100, since these are not empty,... The limit we're hitting is probably something like "Maximum API request size." at `10 MiB` see:
https://cloud.google.com/datastore/docs/concepts/limits

Reply:
We can also make some estimate of the size of these entities..

Do they include the entire pana report?

Reply:
I think that this batching utility should (eventually) live in `shared/datastore.dart` and should estimate the request size, however, I'd defer that to a later point.